app/models/hike.rb

```diff
@@ -47,16 +47,9 @@ def distance_description
     end
   end
 
-  def simplified_waypoints(array)
-    simplified = []
-    array.each_with_index do |point,i|
-      simplified << point if i % 2 == 0
-    end
-    if simplified.length > 300
-      return simplified_waypoints(simplified)
-    else
-      simplified
-    end
+  def simplified_waypoints(waypoints)
+    return waypoints if waypoints.length < 600
+    simplified_waypoints( waypoints.each_slice(2).map(&:first) )
```

Comment:
you could avoid the recursion by using a step function
` (1..waypoints.size).step([1, waypoints.size / 600.0].max).map { |i| waypoints[i.floor - 1] }`
This would yield equidistant samples, up to 600 samples.